### PR TITLE
Print function

### PR DESCRIPTION
**Pull request by NateRiz**
Needed a way to view the programs for debugging / analyzing. Perhaps we can add something that allows for levels / nested instructions as well as an easy-to-read print function similar to sgp1. Let me know if there's something that needs to change.

### CODE REVIEW

File: source/hardware/SignalGP/utils/LinearFunctionsProgram.h [hunk 1]
```diff
@@ -243,7 +248,18 @@ namespace sgp {
...
 
-    // todo - prints
+    // todo - support tabbing/levels for block type instructions
+    // todo - easier to read print without tags
+    template<typename HARDWARE_T, typename INST_PROPERTY_T>
+    void Print(std::ostream& out, const InstructionLibrary<HARDWARE_T, inst_t, INST_PROPERTY_T>& ilib) const{
+      for (size_t i = 0; i < GetSize(); ++i){
+        // Skip the last tag so we dont get an extra delimeter
+        std::copy(program[i].GetTags().begin(), program[i].GetTags().end() - 1, std::ostream_iterator<tag_t>(out, "\n"));
+        out << program[i].GetTags().back() << " Fn-"<< i+1 << "\n";
```
Comment (amlalejini) [inline source/hardware/SignalGP/utils/LinearFunctionsProgram.h:258]: why start function labels at 1 as opposed to 0?

**Review by amlalejini — CHANGES_REQUESTED** [review]
This pull request adds (much needed) Print functions to the LinearProgram and LinearFunctionsProgram classes. 

One or two minor changes requested, but otherwise looks great!

Changes:
- See comment on function labeling
- Can you add a comment to the relevant Print functions that document the format of the print out? e.g., tags are printed one-per-line, etc.

File: source/hardware/SignalGP/utils/LinearFunctionsProgram.h (hunk 1 above)
Comment (NateRiz) [reply source/hardware/SignalGP/utils/LinearFunctionsProgram.h:258]: mostly cosmetic, Ill change it to 0-based

**Comment by NateRiz** [comment]
Added comments & changed to 0-based idx

**Review by amlalejini — APPROVED** [review]

**Pull request merged**